rotkehlchen/inquirer.py

```diff
@@ -500,6 +516,161 @@ def find_uniswap_v2_lp_price(
+                    f'fetch price for {token}.',
+                )
+                return None
+            prices.append(price)
+
+        # Query virtual price of LP share and balances in the pool for each token
+        contract = EthereumContract(
+            address=pool.pool_address,
+            abi=CURVE_POOL_ABI,
+            deployed_block=0,
+        )
+        call = [(pool.pool_address, contract.encode(method_name='get_virtual_price'))]
+        call += [
+            (pool.pool_address, contract.encode(method_name='balances', arguments=[i]))
+            for i in range(len(pool.assets))
+        ]
+        output = multicall_2(
+            ethereum=self._ethereum,
+            require_success=False,
+            calls=call,
+        )
+
+        # Check that the output has the correct structure
+        if not all([len(call_result) == 2 for call_result in output]):
+            log.debug(
+                f'Failed to query contract methods while finding curve pool price. '
+                f'Not every outcome has length 2. {output}',
+            )
+            return None
+        # Check that all the request where successful
+        if not all([contract_output[0] for contract_output in output]):
+            log.debug(f'Failed to query contract methods while finding curve price. {output}')
+            return None
+        # Deserialice information obtained in the multicall execution
+        data = []
+        virtual_price_decoded = contract.decode(output[0][1], 'get_virtual_price')
+        # Verify that the output from the contrat call is correct
+        if (
+            not isinstance(virtual_price_decoded, tuple) or
+            len(virtual_price_decoded) != 1 or
+            not isinstance(virtual_price_decoded[0], int)
+        ):
+            log.debug(f'Failed to decode get_virtual_price while finding curve price. {output}')
+            return None
+        data.append(FVal(virtual_price_decoded[0]))
+        for i in range(len(pool.assets)):
+            amount_decoded = contract.decode(output[i + 1][1], 'balances', arguments=[i])
+            # Verify that the contract decoded the output correctly
+            if (
+                not isinstance(amount_decoded, tuple) or
+                len(amount_decoded) != 1 or
+                not isinstance(amount_decoded[0], int)
+            ):
+                log.debug(f'Failed to decode balances {i} while finding curve price. {output}')
+                return None
+            amount = amount_decoded[0]
+            normalized_amount = token_normalized_value_decimals(amount, tokens[i].decimals)
+            data.append(normalized_amount)
+
+        # Prices and data should verifty this relation for the following operations
+        if len(prices) != len(data) - 1:
+            log.debug(
+                f'Lenght of prices {len(prices)} does not match len of data {len(data)} '
+                f'while querying curve pool price.',
+            )
```

Comment:
```suggestion
            log.debug(
                f'Length of prices {len(prices)} does not match len of data {len(data)} '
                f'while querying curve pool price.',
            )
            return None
```

Shouldn't you also return None here? Notice I changed typo too. Lenght -> Length.

Additionally in all of these places where you do error handling it's a better idea to log an error, and do that via the msg_aggregator, so that the user knows something is wrong and knows to check the logs and send it to us.

Returning None and simply logging in the logs will be very easy to miss. Please do the suggested change everywhere where you log debug/error here.


... unless returning None will end up showing some kind of message to the user, and tell them to check the logs, in which case it's fine. Is it?

Reply:
I've added a check for all this methods and now if any of those returns none a message is returned to the user